test/CMakeLists.txt

```diff
@@ -65,45 +73,46 @@ if( NOT EXISTS ${LLHTTP_DIR}/src/llhttp.c )
 endif()
 
 # ============================ Test configuration ==============================
-
-# Define a CMock resource path.
-set( CMOCK_DIR ${MODULE_ROOT_DIR}/test/unit-test/CMock CACHE INTERNAL "CMock library source directory." )
-
-# Include CMock build configuration.
-include( unit-test/cmock_build.cmake )
-
-# Check if the CMock source directory exists, and if not present, clone the submodule
-# if BUILD_CLONE_SUBMODULES configuration is enabled.
-if( NOT EXISTS ${CMOCK_DIR}/src )
-    # Attempt to clone CMock.
-    if( ${BUILD_CLONE_SUBMODULES} )
-        clone_cmock()
-    else()
-        message( FATAL_ERROR "The required submodule CMock does not exist. Either clone it manually, or set BUILD_CLONE_SUBMODULES to 1 to automatically clone it during build." )
+if( UNITTEST )
```

Comment:
Will this repo need the same change to the CI build as [coreMQTT's ci.yml file](https://github.com/FreeRTOS/coreMQTT/pull/276/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f)?